mne/utils/config.py

```diff
@@ -218,9 +225,54 @@ def set_memmap_min_size(memmap_min_size):
 )
 
 
+@contextlib.contextmanager
+def _open_lock(path, *args, **kwargs):
+    """
+    Context manager that opens a file with an optional file lock.
+
+    If the `filelock` package is available, a lock is acquired on a lock file
+    based on the given path (by appending '.lock').
+
+    Otherwise, a null context is used. The path is then opened in the
+    specified mode.
+
+    Parameters
+    ----------
+    path : str
+        The path to the file to be opened.
+    *args, **kwargs : optional
+        Additional arguments and keyword arguments to be passed to the
+        `open` function.
+
+    """
+    filelock = _soft_import(
+        "filelock", purpose="parallel config set and get", strict=False
+    )
+
+    lock_context = contextlib.nullcontext()  # default to no lock
+
+    if filelock is not None:
+        lock_path = f"{path}.lock"
+        try:
+            from filelock import FileLock
+
+            lock_context = FileLock(lock_path, timeout=5)
+            lock_context.acquire()
+        except TimeoutError:
+            lock_context = contextlib.nullcontext()
```

Comment:
Hmmm.... if a user ends up with a `.mne-python.json.lock` file "stuck" on their system (like it didn't clean up properly), then they'll silently hit a 5-second delay every time we `get_config` any value, right? That would be bad as it's a silent, hard-to-track-down slowdown. So we should probably add something like
```suggestion
        except TimeoutError:
            warn(
                "Could not acquire lock file after 5 seconds, consider deleting it "
                f"if you know the corresponding file is usable:\n{lock_path}"
            )
            lock_context = contextlib.nullcontext()
```